pulp_container/app/token_verification.py

```diff
@@ -116,6 +113,26 @@ def authenticate_header(self, request):
         return authenticate_string
 
 
+class RegistryPermission(BasePermission):
+    """
+    Permission class to determine permissions based on the request user.
+    """
+
+    message = "Access to the requested resource is not authorized."
+
+    def has_permission(self, request, view):
+        """
+        Decide upon permission based on user.
+        """
+        # TODO RBAC goes here
+        if request.user.is_staff:
```

Comment:
I have tested this pr, user here is anonymours no matter what i am doing
1. login into the registry permits to login even with wrong credentials
2. if i'm accessing api directly this codepath is niot executed and i can perform get/post/put call without any auth provided.

Reply:
You are right. As long as we don't force "/v2/" to need credentials, podman will never try to provide them.
But we break anonymous pull if we do, because he will still try to find that and fallback to /v1ping/ instead.
This needs more investigation.